tests/templates/test_subroutines/test_prepselprep.py

```diff
@@ -29,7 +29,7 @@
         (qml.ops.LinearCombination([0.25, 0.75], [qml.Z(2), qml.X(1) @ qml.X(2)]), [0]),
         (qml.ops.LinearCombination([-0.25, 0.75j], [qml.Z(3), qml.X(2) @ qml.X(3)]), [0, 1]),
         (
-            qml.ops.LinearCombination([-0.25 + 0.1j, 0.75j], [qml.Z(4), qml.X(4) @ qml.X(4)]),
+            qml.ops.LinearCombination([-0.25 + 0.1j, 0.75j], [qml.Z(4), qml.X(4) @ qml.X(5)]),
```

Comment:
why this one?

Reply:
Because `SProd.terms()` now calls `Prod.terms()` here, which simplifies `qml.X(4) @ qml.X(4)` to `I()`, which broke `flatten`/`unflatten` of `PrepSelPrep`. This is a bug with `PrepSelPrep` that Guillermo is working on. For now I'm making this change to unblock my PR and get it merged by the end of Friday, because I'm going on vacation next week.

Reply:
Awesome, thanks for the context!